bot/cogs/extensions.py

```diff
@@ -0,0 +1,231 @@
+EXTENSIONS = frozenset(
+    ext.name
+    for ext in iter_modules(("bot/cogs",), "bot.cogs.")
+    if ext.name[-1] != "_"
+)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(Bot.load_extension)
+    UNLOAD = functools.partial(Bot.unload_extension)
+    RELOAD = functools.partial(Bot.reload_extension)
+
+
+class Extension(commands.Converter):
+    """
+    Fully qualify the name of an extension and ensure it exists.
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        """Fully qualify the name of an extension and ensure it exists."""
+        # Special values to reload all extensions
+        if ctx.command.name == "reload" and (argument == "*" or argument == "**"):
+            return argument
+
+        argument = argument.lower()
+
+        if "." not in argument:
+            argument = f"bot.cogs.{argument}"
+
+        if argument in EXTENSIONS:
+            return argument
+        else:
+            raise commands.BadArgument(f":x: Could not find the extension `{argument}`.")
+
+
+class Extensions(commands.Cog):
+    """Extension management commands."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @group(name="extensions", aliases=("ext", "exts", "c", "cogs"), invoke_without_command=True)
+    async def extensions_group(self, ctx: Context) -> None:
+        """Load, unload, reload, and list loaded extensions."""
+        await ctx.invoke(self.bot.get_command("help"), "extensions")
+
+    @extensions_group.command(name="load", aliases=("l",))
+    async def load_command(self, ctx: Context, extension: Extension) -> None:
+        """Load an extension given its fully qualified or unqualified name."""
+        msg, _ = self.manage(extension, Action.LOAD)
+        await ctx.send(msg)
+
+    @extensions_group.command(name="unload", aliases=("ul",))
+    async def unload_command(self, ctx: Context, extension: Extension) -> None:
+        """Unload a currently loaded extension given its fully qualified or unqualified name."""
+        if extension in UNLOAD_BLACKLIST:
+            msg = f":x: The extension `{extension}` may not be unloaded."
+        else:
+            msg, _ = self.manage(extension, Action.UNLOAD)
+
+        await ctx.send(msg)
+
+    @extensions_group.command(name="reload", aliases=("r",))
+    async def reload_command(self, ctx: Context, *extensions: Extension) -> None:
+        """
+        Reload extensions given their fully qualified or unqualified names.
+
+        If an extension fails to be reloaded, it will be rolled-back to the prior working state.
+
+        If `*` is given as the name, all currently loaded extensions will be reloaded.
+        If `**` is given as the name, all extensions, including unloaded ones, will be reloaded.
```

Comment:
The asterisks inside the inline code blocks need to be escapes but I don't know how. The help command italicises the docstring, so having these asterisks messes things up.

Reply:
I've tried a couple of things, but I have no idea how to do this either. We may need to break with our usual style here and go with another solution than an inline code block. I think it may just be impossible for now.

Reply:
An alternative style would be to just have it as normal text. Escaping it seems to work then, but I will confirm tomorrow.

Reply:
Normal text sounds fine to me. (As far as I can tell, escaping then works as expected.)

Reply:
If you add a space between the \` and \*, ie \` *`, it will make the astericks in the codeblock, but the whole description won't be italicised. Other than that I couldn't figure it out either

Reply:
Went with normal text.